### DISCUSSION TITLE
ImageAnnotations erroneously accumulate in Studio

### DISCUSSION THREAD

**Discussion by willcray** [foxglove/community#1022] [Bugs]
Hello, I started overlaying ImageAnnotations onto my raw RGB images. However, when a new RGB image and image annotation are loaded, the old image annotation remains. We use a pub-sub, Redis-like communication bus. I would expect Foxglove to only render the most recent object from each topic, just as it does for all of our other topics.

I see the option in Studio to synchronize image annotations with an image. However, we currently don't have the infrastructure to synchronize them.

Any ideas?

**Comment by defunctzombie** [comment]
Do you have your own implementation of the foxglove websocket protocol that interfaces with your pub-sub bus? The first step here is to capture the data from your pub-sub into an mcap file so we can see what you see.

**Comment by jtbandes** [comment]
@willcray it looks like you marked this discussion as closed. Are you still having a problem with this or is it resolved?

**Reply by willcray** [reply]
Hey, we found the issue--it was on our end. Sorry to bother!
